zetaclient/chains/bitcoin/observer/inbound.go

```diff
@@ -472,3 +472,173 @@ func GetBtcEvent(
 	}
 	return nil, nil
 }
+
+// GetBtcEventWithWitness either returns a valid BTCInboundEvent or nil.
+// This method supports data with more than 80 bytes by scanning the witness for possible presence of a tapscript.
+// It will first prioritize OP_RETURN over tapscript.
+func GetBtcEventWithWitness(
+	client interfaces.BTCRPCClient,
+	tx btcjson.TxRawResult,
+	tssAddress string,
+	blockNumber uint64,
+	logger zerolog.Logger,
+	netParams *chaincfg.Params,
+	depositorFee float64,
+) (*BTCInboundEvent, error) {
+	if len(tx.Vout) < 1 {
+		logger.Debug().Msgf("no output %s", tx.Txid)
+		return nil, nil
+	}
+	if len(tx.Vin) == 0 { // should never happen
+		logger.Debug().Msgf("no input found for inbound: %s", tx.Txid)
+		return nil, nil
+	}
+
+	if !isValidRecipient(tx.Vout[0].ScriptPubKey.Hex, tssAddress, netParams, logger) {
+		logger.Debug().Msgf("irrelevant recipient %s for inbound: %s", tx.Vout[0].ScriptPubKey.Hex, tx.Txid)
+		return nil, nil
+	}
+
+	isAmountValid, amount := isValidAmount(tx.Vout[0].Value, depositorFee)
+	if !isAmountValid {
+		logger.Info().
+			Msgf("GetBtcEventWithWitness: btc deposit amount %v in txid %s is less than depositor fee %v", tx.Vout[0].Value, tx.Txid, depositorFee)
+		return nil, nil
+	}
+
+	var memo []byte
+	if candidate := tryExtractOpRet(tx, logger); candidate != nil {
+		memo = candidate
+		logger.Debug().Msgf("GetBtcEventWithWitness: found OP_RETURN memo %s in tx %s", hex.EncodeToString(memo), tx.Txid)
+	} else if candidate = tryExtractInscription(tx, logger); candidate != nil {
+		memo = candidate
+		logger.Debug().Msgf("GetBtcEventWithWitness: found inscription memo %s in tx %s", hex.EncodeToString(memo), tx.Txid)
+	} else {
+		return nil, nil
+	}
```

Comment:
Would be great to add comment, what we're doing in this block

Reply:
Added!